ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
@lukewagner, all sounds very good to me. I've always thought of these options as "lists with a length limit" (and the design doc actually describes them as such). Reverting them to actual lists in ml-proto would perhaps make that clearer, and as you say, would match the binary encoding that we may be converging on.

@JSStats, once we add multi-value support post-MVP, I agree that we should add a `tuple` or `values` operator. Re stack machine: yes indeed, and that would be one possible way to describe the semantics. There is no problem with that, because the language is typed, so this stack is guaranteed to always be manipulated consistently. Basically, the benefit of making tuples/multi-values second-class is that both interpretations/implementation strategies are interchangeable, because a program cannot observe the difference. In fact, maintaining that equivalence is a basic litmus test for getting the multi-value semantics right.